src/backend/base/langflow/components/vectorstores/OpenSearch.py

```diff
@@ -0,0 +1,292 @@
+                        embedding_function=self.embedding,
+                        opensearch_url=self.opensearch_url,
+                        http_auth=(self.username, self.password),
+                        use_ssl=self.use_ssl,
+                        verify_certs=self.verify_certs,
+                        ssl_assert_hostname=False,
+                        ssl_show_warn=False,
+                    )
+                    return opensearch
+                except Exception as e:
+                    logger.error(f"Failed to connect to OpenSearchVectorSearch instance: {str(e)}")
+                    raise RuntimeError(
+                        "Error connecting to OpenSearchVectorSearch instance. Check your connection and credentials."
+                    ) from e
+        except Exception as e:
+            logger.error(
+                f"Unexpected error in build_vector_store, checkout if the username and the password were correctly set: {str(e)}"
+            )
+            raise RuntimeError(
+                "An unexpected error occurred while building the vector store, checkout if the username and the password were correctly set."
+            ) from e
+
+    def _add_documents_to_vector_store(self, vector_store: "OpenSearchVectorSearch") -> None:
+        """
+        Adds documents to the Vector Store.
+        """
+        if not self.ingest_data:
+            self.status = ""
+            return
+
+        documents = []
+        for _input in self.ingest_data or []:
+            if isinstance(_input, Data):
+                documents.append(_input.to_lc_document())
+            else:
+                error_message = "Vector Store Inputs must be Data objects."
+                logger.error(error_message)
+                raise ValueError(error_message)
+
+        if documents and self.embedding is not None:
+            logger.debug(f"Adding {len(documents)} documents to the Vector Store.")
+            try:
+                vector_store.add_documents(documents)
+            except Exception as e:
+                error_message = f"Error adding documents to Vector Store: {str(e)}"
+                logger.error(error_message)
+                logger.error(f"Traceback: {traceback.format_exc()}")
+                raise Exception(error_message)
+        else:
+            logger.debug("No documents to add to the Vector Store.")
+
+    def search(self, query: Optional[str] = None) -> List[Dict[str, Any]]:
+        """
+        Search for similar documents in the vector store or retrieve all documents if no query is provided.
+        """
+        try:
+            vector_store = self.build_vector_store()
+            search_kwargs = {"k": self.number_of_results, "score_threshold": self.search_score_threshold}
+
+            if query:
+                search_type = self.search_type.lower()
+                if search_type == "similarity":
+                    results = vector_store.similarity_search_with_score(query, **search_kwargs)
```

Comment:
I don't believe this should be the default. Langchain's pattern is to handle three main types: `similarity`, `similarity_score_threshold`, and `mmr`. The second handles the `score`. 

https://github.com/langchain-ai/langchain/blob/5346c7b27ec50ddf156c6aff15854185a62a1af4/libs/core/langchain_core/vectorstores/base.py#L315

Unless there's a particular difference using OpenSearch, is there any reason we can't use the default search impl?

Reply:
It was wrong, I also fixed it to match the default implementation..